### PR TITLE
A4A: Add Marketplace dedicated sidebar section.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request adds a new Marketplace sidebar section to match the updated Information Architecture. The updated architecture now has a dedicated Products page and Hosting page.
Screenshot 2024-03-14 at 4 15 39 PM
Screenshot 2024-03-14 at 4 15 47 PM

Closes https://github.com/Automattic/jetpack-genesis/issues/282

## Proposed Changes

* Add Marketplace sidebar navigation menu.
* Add Hosting overview placeholder page.
* Update Marketplace controller to include 'Products' and 'Hosting' routes.
* Update Layout to support spacing with top border. Additionally, I added new 'compact' props for different spacing sizes. This will be used later on the checkout and Pressable/Wordpress.com site page.

## Testing Instructions
* Switch branch: `git checkout add/a4a-marketplace-sidebar-section`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/
* Click the 'Marketplace' sidebar menu item.
* Confirm that you are redirected to the Products page. 
* Click 'Hosting' on the sidebar menu item.
* Confirm that you are redirected to the Hosting page.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/layout/index.tsx [hunk 1]
```diff
@@ -12,20 +12,23 @@ type Props = {
...
+	wide,
+	withBorder,
+	compact,
 	sidebarNavigation,
 }: Props ) {
 	return (
 		<Main
 			className={ classNames( 'a4a-layout', className, {
 				'is-with-border': withBorder,
+				'is-compact': compact,
```
Comment (jkguidaven) [inline client/a8c-for-agencies/components/layout/index.tsx:31]: The spacing for the **Checkout** and **Pressable/Wordpress.com** hosting page has different spacing. By using a flag, we can determine if we need a spacing of 48px or 28px.

File: client/a8c-for-agencies/sections/marketplace/controller.tsx [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
-export const marketplaceContext: Callback = ( context, next ) => {
+export const marketplaceContext: Callback = () => {
+	page.redirect( '/purchases/licenses' );
+};
+
+export const marketplaceProductsContext: Callback = ( context, next ) => {
 	const { site_id, product_slug } = context.query;
-	context.secondary = <MainSidebar path={ context.path } />;
+	context.secondary = <MarketplaceSidebar path={ context.path } />;
 	context.primary = <IssueLicense siteId={ site_id } suggestedProduct={ product_slug } />;
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/controller.tsx:17]: In the future, we plan to change the name of the `IssueLicense` component to `ProductsOverview` to align with the route naming convention.

File: client/a8c-for-agencies/components/sidebar-menu/lib/constants.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ export const A4A_SITES_LINK_NEEDS_ATTENTION = '/sites?issue_types=all_issues';
 export const A4A_SITES_LINK_FAVORITE = '/sites/favorite';
 export const A4A_PLUGINS_LINK = '/plugins';
 export const A4A_MARKETPLACE_LINK = '/marketplace';
+export const A4A_MARKETPLACE_PRODUCTS_LINK = '/marketplace/products';
+export const A4A_MARKETPLACE_HOSTING_LINK = '/marketplace/hosting';
```
Comment (yashwin) [inline client/a8c-for-agencies/components/sidebar-menu/lib/constants.ts]: [Non-blocker] Can we use `${ A4A_MARKETPLACE_LINK  }`?

File: client/a8c-for-agencies/sections/marketplace/index.tsx [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
+	assignLicenseContext,
+	marketplaceContext,
+	marketplaceHostingContext,
+	marketplaceProductsContext,
+} from './controller';
 
 export default function () {
 	page( '/marketplace', marketplaceContext, makeLayout, clientRender );
+	page( '/marketplace/products', marketplaceProductsContext, makeLayout, clientRender );
+	page( '/marketplace/hosting', marketplaceHostingContext, makeLayout, clientRender );
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/index.tsx]: [Non-blocker] Can we get the links from the constant file?

**Review by yashwin — COMMENTED** [review]
Looks good overall.

Can we use the right icons as per the design?

Screenshot 2024-03-14 at 1 16 01 PM

**Comment by jkguidaven** [comment]
@yashwin, thanks for the review. I've made some adjustments to address your comments. Can you have another look? 🙏

File: client/a8c-for-agencies/sections/marketplace/controller.tsx [hunk 5]
```diff
@@ -1,23 +1,35 @@
...
 import getSites from 'calypso/state/selectors/get-sites';
-import MainSidebar from '../../components/sidebar-menu/main';
+import MarketplaceSidebar from '../../components/sidebar-menu/marketplace';
 import AssignLicense from './assign-license';
+import HostingOverview from './hosting-overview';
 import IssueLicense from './issue-license';
 
-export const marketplaceContext: Callback = ( context, next ) => {
+export const marketplaceContext: Callback = () => {
+	page.redirect( '/purchases/licenses' );
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/controller.tsx]: Sorry, just realised this should be `/marketplace/products`?

Comment (jkguidaven) [reply client/a8c-for-agencies/sections/marketplace/controller.tsx]: Good catch. This has been fixed 👍

**Review by yashwin — COMMENTED** [review]
Thanks for addressing my comments!

LGTM now :shipit:

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11603627

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11603627).

**Pull request merged**